src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -5153,6 +5154,25 @@ class LockTableStatementSegment(BaseSegment):
     )
 
 
+class ClusterStatementSegment(BaseSegment):
+    """A `CLUSTER` statement.
+
+    https://www.postgresql.org/docs/current/sql-cluster.html
+    """
+
+    type = "cluster_statement"
+    match_grammar = Sequence(
+        "CLUSTER",
+        Ref.keyword("VERBOSE", optional=True),
+        Ref("ObjectReferenceSegment", optional=True),
+        OneOf(
+            Sequence("USING", Ref("ObjectReferenceSegment")),
+            Sequence("ON", Ref("ObjectReferenceSegment")),
```

Comment:
As this is for old old back-compatibility, it may be cleaner to do `OneOf( table [using index], index on table)`